Assets/Scripts/Utils/SteamManager.cs

```diff
@@ -55,11 +75,59 @@ private void Start()
     {
         if (!isSteamInitialized)
             return;
+
         lobbyCreated = Callback<LobbyCreated_t>.Create(OnlobbyCreated);
         lobbyEnter = Callback<LobbyEnter_t>.Create(OnLobbyEnter);
         joinRequest = Callback<GameLobbyJoinRequested_t>.Create(OnJoinRequest);
+
+        RequestStats();
     }
 
+    private void RequestStats()
+    {
+        if (!isSteamInitialized)
+            return;
+
+        SteamUserStats.RequestCurrentStats();
+    }
+
+    #region Achievements
+
+    private readonly Dictionary<AchievementType, string> achievementNames = new()
+    {
+        { AchievementType.DiscoInferno, "WEAPON_DISCO_INFERNO" },
+        { AchievementType.WireFraud, "WEAPON_WIRE_FRAUD" },
+        { AchievementType.SchizoidMan, "WEAPON_SCHIZOID_MAN" },
+        { AchievementType.BlueHat, "WEAPON_BLUE_HAT" },
+        { AchievementType.BlackHat, "WEAPON_BLACK_HAT" },
+        { AchievementType.EagleEyed, "WEAPON_EAGLE_EYED" },
+        { AchievementType.PogoStick, "WEAPON_POGO_STICK" },
+        { AchievementType.Skater, "WEAPON_SKATER" },
+        { AchievementType.SprayNPray, "WEAPON_SPRAY_N_PRAY" },
+        { AchievementType.PingPonginator, "WEAPON_PING_PONG" },
+        { AchievementType.Flamethrower, "WEAPON_FLAMETHROWER" },
+        { AchievementType.OrbitalTrashCannon, "WEAPON_ORBITAL_TRASH_CANNON" },
+        { AchievementType.ItalianPlumber, "WEAPON_ITALIAN_PLUMBER" }
+    };
+
+    public void UnlockAchievement(AchievementType type)
+    {
+        if (type is AchievementType.None || !achievementNames.TryGetValue(type, out var name))
```

Comment:
This function needs to return if steammanger is not initialized, otherwise the game will break if players get a secret name with steam being active 
![image](https://github.com/hackerspace-ntnu/Red-Planet-Rampage/assets/54811121/267b9377-39d4-472f-8b97-f498d5236ec1)